src/Build/BackEnd/Node/ServerNodeBuildCommand.cs

```diff
@@ -49,20 +52,50 @@ internal sealed class ServerNodeBuildCommand : INodePacket
         /// </summary>
         public CultureInfo UICulture => _uiCulture;
 
+        /// <summary>
+        /// Buffer width of destination Console.
+        /// Console loggers are supposed, on Windows OS, to be wrapping to avoid output trimming.
+        /// -1 console buffer width can't be obtained.
+        /// </summary>
+        public int ConsoleBufferWidth => _consoleBufferWidth;
+
+        /// <summary>
+        /// True if console output accept ANSI colors codes.
+        /// False if output is redirected to non screen type such as file or nul.
+        /// </summary>
+        public bool AcceptAnsiColorCodes => _acceptAnsiColorCodes;
+
+        /// <summary>
+        /// True if console output is screen. It is expected that non screen output is post-processed and often does not need wrapping and coloring.
+        /// False if output is redirected to non screen type such as file or nul.
+        /// </summary>
+        public bool ConsoleIsScreen => _consoleIsScreen;
+
+        /// <summary>
+        /// Background color of client console, -1 if not detectable
+        /// </summary>
+        public ConsoleColor ConsoleBackgroundColor => _consoleBackgroundColor;
+
         /// <summary>
         /// Private constructor for deserialization
         /// </summary>
         private ServerNodeBuildCommand()
         {
         }
 
-        public ServerNodeBuildCommand(string commandLine, string startupDirectory, Dictionary<string, string> buildProcessEnvironment, CultureInfo culture, CultureInfo uiCulture)
+        public ServerNodeBuildCommand(string commandLine, string startupDirectory, Dictionary<string, string> buildProcessEnvironment, CultureInfo culture, CultureInfo uiCulture,
+            int consoleBufferWidth, bool acceptAnsiColorCodes, bool consoleIsScreen, ConsoleColor consoleBackgroundColor)
         {
             _commandLine = commandLine;
             _startupDirectory = startupDirectory;
             _buildProcessEnvironment = buildProcessEnvironment;
             _culture = culture;
             _uiCulture = uiCulture;
+
+            _consoleBufferWidth = consoleBufferWidth;
+            _acceptAnsiColorCodes = acceptAnsiColorCodes;
+            _consoleIsScreen = consoleIsScreen;
+            _consoleBackgroundColor = consoleBackgroundColor;
```

Comment:
very minor: any value in validating .ctor params?
All new code params are value types, so no risk of nulls or things in BAD state.

Reply:
I believe it would have negligible affect on code quality